core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -354,21 +354,26 @@ private void applyConfiguration(CreateContainerCmd createCommand) {
 
         Set<Link> allLinks = new HashSet<>();
         Set<String> allLinkedContainerNetworks = new HashSet<>();
+
+        List<com.github.dockerjava.api.model.Container> containers = dockerClient.listContainersCmd().exec();
+
+        List<InspectContainerResponse> inspectContainerResponses = containers.stream()
+                .map(container -> dockerClient.inspectContainerCmd(container.getId()).exec())
+                .collect(Collectors.toList());
+
         for (Map.Entry<String, LinkableContainer> linkEntries : linkedContainers.entrySet()) {
 
             String alias = linkEntries.getKey();
             LinkableContainer linkableContainer = linkEntries.getValue();
 
-            Set<Link> links = dockerClient.listContainersCmd().exec().stream()
-                    .filter(container -> container.getNames()[0].endsWith(linkableContainer.getContainerName()))
-                    .map(container -> new Link(container.getNames()[0], alias))
+            Set<Link> links = inspectContainerResponses.stream()
+                    .filter(inspectResponse -> inspectResponse.getName().endsWith(linkableContainer.getContainerName()))
+                    .map(inspectResponse -> new Link(inspectResponse.getName(), alias))
                     .collect(Collectors.toSet());
             allLinks.addAll(links);
 
-            boolean linkableContainerIsRunning = dockerClient.listContainersCmd().exec().stream()
```

Comment:
Would be happy to do so. This is what I was trying to do here - I extract the list cmd to a single place out of the loops and then re-used it.